### DISCUSSION TITLE
Use sub-folders as sub-topics with correct ordering

### DISCUSSION THREAD

**Discussion by buhtz** [mkdocs/mkdocs#3717] [Q&A]
I try not to use the `nav:` section but want to make MkDocs determine the sections by itself.

```
├── 1000.quick-start.md
├── 2000.main-window.md
├── 3000.settings.md
├── 4000.snapshots-dialog.md
├── 4500.sub-topic
│   ├── 1000.foobar.md
│   └── index.md
├── 5000.log.md
├── 6000.additional.md
└── index.md
```

As you can see I do use numeric prefix in the filenames to manipulate the ordering of the sections in the resulting navigation bar.
This works well for the markdown files but not for the sub folder. The folder `4500.sub-topic` is not rendered between "Snapshots Dialog" and "Log" but at the end of the navigation entries.

And another problem is that the navigation entry is named "Sub Topic" instead of using the first level header in the `4500.sub-topic/index.md` file.

Is there a way to handle this without using `nav:` section in the configuration file?

**Comment by Andre601** [comment]
This is just how folders and files are sorted from what I know.
You can see the same behaviour on windows, where folders are listed before or after files, but never in-between.

**Reply by buhtz** [reply]
I am aware of that. But MkDocs itself could sort files and folders its own way. ;)
The feature of automatic determining the navigation bar entries by filenames of md files doesn't make much sense when folders handled differently.

**Reply by pawamoy** [reply]
> I am aware of that. But MkDocs itself could sort files and folders its own way. ;)

It could, but different users want different things, so MkDocs just lets the system decide on the ordering, as it's impossible to satisfy everyone *by default*.

If you think there's a problem (or a regression) in how files and folders are handled, please open a new issue :relaxed: 

Maybe you can change the order of entries with a [custom hook](https://www.mkdocs.org/user-guide/configuration/#hooks) that reorders the nav (in the `on_nav` event probably).

**Reply by Andre601** [reply]
I also feel like making a custom hook (or maybe even a plugin if there is a demand for this) is the best aproach here.

**Reply by kamilkrzyskow** [reply]
How about using the [awesome pages MkDocs plugin](https://github.com/lukasgeiter/mkdocs-awesome-pages-plugin/)
There are settings for ordering and sorting, which could help in this scenario 🤔

**Comment by buhtz** [comment]
Thanks for your insights. These are all nice options.

But I realized that mkdocs would warn me if I have md files that are not included in the "nav:" section. This is enough for me.
